src/lib.rs

```diff
@@ -20,6 +20,7 @@ impl<T> ServerInput for T where T: BufRead + Read {}
 pub struct Server {
     reader: Box<dyn ServerInput>,
     writer: Box<dyn Write>,
+    pub disable_folding: bool,
```

Comment:
why does this need to be added to server, should we just pass it through and not hold on it?